From 765aac5d884b362596b3e15fb1d9bb8f7f8be76c Mon Sep 17 00:00:00 2001
From: Joe Conley <joe.conley@lairdtech.com>
Date: Thu, 25 Feb 2016 16:50:56 -0500
Subject: [PATCH] laird: ebtables: only set BOOTP relay address if it is on the
 outbound interface

If no address is set, BOOTP relay address will be 0.0.0.0
All other interface addresses are ignored

bug 9006
---
 net/bridge/netfilter/ebt_arpnat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/bridge/netfilter/ebt_arpnat.c b/net/bridge/netfilter/ebt_arpnat.c
index c836bf94cb7a..cb5f92f37a82 100644
--- a/net/bridge/netfilter/ebt_arpnat.c
+++ b/net/bridge/netfilter/ebt_arpnat.c
@@ -549,7 +549,7 @@ static unsigned int ebt_target_arpnat(struct sk_buff *pskb, const struct xt_acti
 				if (uh->dest == htons(67) || uh->dest == htons(68) )
 				{
 					//do something illegal for BOOTP
-					uint32_t giaddr = inet_select_addr(out_br_port->br->dev, iph->daddr, RT_SCOPE_LINK);
+					uint32_t giaddr = inet_confirm_addr(dev_net(out), __in_dev_get_rcu(out), 0, 0, RT_SCOPE_LINK);
 					uint32_t* giaddrp = (uint32_t*)(((uint8_t*)uh) + sizeof(*uh) + GIADDR_OFFSET);
 					uint8_t *chaddrp = (uint8_t*)(((uint8_t*)uh) + sizeof(*uh) + CHADDR_OFFSET);
 					uint32_t ihl = iph->ihl << 2;
-- 
2.11.0

